src/libstd/io/mod.rs

```diff
@@ -942,6 +942,57 @@ impl<'a> IoSliceMut<'a> {
     pub fn new(buf: &'a mut [u8]) -> IoSliceMut<'a> {
         IoSliceMut(sys::io::IoSliceMut::new(buf))
     }
+
+    /// Advance the internal cursor of the slice.
+    ///
+    /// # Notes
+    ///
+    /// Elements in the slice may be modified.
```

Comment:
This might need expanding.

Reply:
Yeah I'd recommend going into detail here and explaining how the vector elements of the slice may be modified to adjust the size as noted by `advance`

Reply:
What do you think of what I've added?